### ISSUE TITLE
Consider using workflow commands (stdout) instead of API calls using token which doesn't work from fork PRs

### ISSUE THREAD

**Issue by jnm2** [sibiraj-s/action-eslint#3]
You can write workflow commands to stdout instead of making web API calls which depend on the token. The token is readonly when triggered for PRs from forks and causes the action to fail (e.g. https://github.com/wearerequired/lint-action/issues/13). With stdout, there is no authentication issue and the performance may even be better.

Example from docs page (https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-a-warning-message):

```
echo "::warning file=app.js,line=1,col=5::Missing semicolon"
```

Both warning and error annotations can be created this way. A GitHub action using this approach: https://github.com/cschleiden/jest-github-actions-reporter

![](https://raw.githubusercontent.com/cschleiden/jest-github-actions-reporter/master/img/annotations.png)

**Comment by moodysalem** [comment]
@samuelmeuli we have a lot of PRs that would benefit from this https://github.com/Uniswap/uniswap-interface/pulls

**Comment by sibiraj-s** [comment]
Sorry for the delayed reply, was occupied heavily. I am not sure if I am in the right context here. 

The Github Token here is used to read/fetch the files changed in the PR using the oktokit api and thats it, Once fetched it then calls eslint installed by the user with the changed files. 

The annotations are caught automatically from the report generated by eslint by problem matchers registered by [setup-node](https://github.com/actions/setup-node/blob/main/.github/eslint-stylish.json).

**Comment by jnm2** [comment]
I see, that makes sense. There were several lint actions that had the issue I described and I got mixed up. Sorry for the confusion!

**Comment by sibiraj-s** [comment]
No issues.
